project/shacl/modos_schema.shacl.ttl

```diff
@@ -136,87 +136,143 @@ modos:MODOCollection a sh:NodeShape ;
+            sh:order 6 ;
+            sh:path schema1:description ],
+        [ sh:datatype xsd:anyURI ;
+            sh:description "A unique identifier for a thing" ;
+            sh:maxCount 1 ;
+            sh:minCount 1 ;
+            sh:nodeKind sh:Literal ;
+            sh:order 4 ;
+            sh:path schema1:identifier ],
+        [ sh:datatype xsd:string ;
+            sh:description "A human-readable name for a thing" ;
+            sh:maxCount 1 ;
+            sh:nodeKind sh:Literal ;
+            sh:order 5 ;
+            sh:path schema1:name ],
         [ sh:class modos:ReferenceGenome ;
             sh:description "Specifies the reference coordinate system used by an omics dataset." ;
             sh:nodeKind sh:IRI ;
             sh:order 3 ;
             sh:path modos:has_reference ],
         [ sh:description "Data/file format associated with a data entity." ;
-            sh:in ( EDAM:format_3462 EDAM:format_1930 EDAM:format_3915 EDAM:format_1929 EDAM:format_3016 EDAM:format_3020 ) ;
+            sh:in ( EDAM:format_3462 EDAM:format_1930 EDAM:format_3915 EDAM:format_1929 EDAM:format_3016 EDAM:format_3020 EDAM:format_3681 ) ;
             sh:maxCount 1 ;
             sh:minCount 1 ;
             sh:order 1 ;
             sh:path modos:data_format ],
-        [ sh:datatype xsd:anyURI ;
-            sh:description "A unique identifier for a thing" ;
+        [ sh:datatype xsd:string ;
+            sh:description "The path to access a resource, on a network or local filesystem. Can be a path relative to the root of the digital object, a URL, or an absolute path." ;
             sh:maxCount 1 ;
             sh:minCount 1 ;
             sh:nodeKind sh:Literal ;
-            sh:order 4 ;
-            sh:path schema1:identifier ],
+            sh:order 0 ;
+            sh:path modos:data_path ],
         [ sh:class modos:Sample ;
             sh:description "Biological sample included or described by a given thing." ;
             sh:nodeKind sh:IRI ;
             sh:order 2 ;
-            sh:path modos:has_sample ],
-        [ sh:datatype xsd:string ;
-            sh:description "The path to access a resource, on a network or local filesystem. Can be a path relative to the root of the digital object, a URL, or an absolute path." ;
-            sh:maxCount 1 ;
-            sh:minCount 1 ;
-            sh:nodeKind sh:Literal ;
-            sh:order 0 ;
-            sh:path modos:data_path ] ;
+            sh:path modos:has_sample ] ;
     sh:targetClass modos:VariantSet .
 
 modos:Assay a sh:NodeShape ;
     sh:closed true ;
     sh:description "A coordinated set of actions designed to generate data from samples." ;
     sh:ignoredProperties ( rdf:type ) ;
-    sh:property [ sh:datatype xsd:string ;
+    sh:property [ sh:datatype xsd:anyURI ;
+            sh:description """Codes describing sample processing, preparation or handling steps.
+The order of the codes should reflect the order in which the steps were performed.
```

Comment:
this is not a thing in SHACL/RDF. Unless you explicitly add triples that specify the order of each node, you cannot say whether CODE123 comes "before" CODE456.

Reply:
Thanks, for now this should be OK (esp. because it's persisted as json, which supports it. If it becomes a need, we'll include ranks/order in the schema.